client/js/components/statement/assessmentTable/Fragment.vue

```diff
@@ -740,25 +734,71 @@ export default {
-      const shouldIgnoreLastClaimed = this.fragment.departmentId === '' && hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee.id === this.currentUserId
+      const shouldIgnoreLastClaimed = this.fragment.departmentId === '' && hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee?.id === this.currentUserId
 
       const assigneeData = {
         fragmentId: this.fragmentId,
         statementId: this.statement.id,
         ignoreLastClaimed: shouldIgnoreLastClaimed,
-        assigneeId: (hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee.id === this.currentUserId ? '' : this.currentUserId),
-        ...((shouldIgnoreLastClaimed === false && this.fragment.assignee.id === this.currentUserId) && { lastClaimed: this.currentUserId })
+        assigneeId: (hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee?.id === this.currentUserId ? '' : this.currentUserId),
+        ...((shouldIgnoreLastClaimed === false && this.fragment.assignee?.id === this.currentUserId) && { lastClaimed: this.currentUserId })
       }
 
       this.setAssigneeAction(assigneeData)
         .then(() => {
           this.updatingClaimState = false
           this.$root.$emit('entity:updated', this.fragment.id, 'fragment')
 
-          if (hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee.id !== this.currentUserId) {
+          if (hasOwnProp(this.fragment.assignee, 'id') && this.fragment.assignee?.id !== this.currentUserId) {
             this.reviewerEditing = false
           }
         })
+    },
+
+    updateField (field, updated) {
+      // If tags are changed, we have to add the tag content to recommendation text field
+      if (field === 'tags') {
+        // This string will concatenate all tags' texts we want to later add to consideration/considerationAdvice
+        let textToBeAdded = ''
+        // First we have to know which field we want to update (which field is visible)
+        let fieldToUpdate = ''
+        if (this.$refs.consideration) {
+          fieldToUpdate = this.$refs.consideration
+        } else if (this.$refs.considerationAdvice) {
+          fieldToUpdate = this.$refs.considerationAdvice
+        }
+
+        this.updateTag(data, fieldToUpdate, textToBeAdded, updated)
+        this.addTagsToConsideration(textToBeAdded, fieldToUpdate)
+      }
+    },
+
+    updateTag (data, fieldToUpdate, textToBeAdded, updated) {
+      // Then we fire a request for each tag - we return an array of promises to wait until all requests are finished
```

Comment:
I'd move this comment to where `updateTag()` is called, because right now it's not clear what the "Then" refers to